packages/core/android/src/main/java/io/sentry/react/RNSentrySDK.java

```diff
@@ -0,0 +1,82 @@
+package io.sentry.react;
+
+import android.content.Context;
+import com.facebook.react.bridge.ReadableMap;
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.android.core.AndroidLogger;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+
+public final class RNSentrySDK {
+  private static final String CONFIGURATION_FILE = "sentry.options.json";
+  private static final String NAME = "RNSentrySDK";
+
+  private static final ILogger logger = new AndroidLogger(NAME);
+  private static final RNSentryStart startSdk = new RNSentryStart();
+
+  private RNSentrySDK() {
+    throw new AssertionError("Utility class should not be instantiated");
+  }
+
+  private static void startWithOptions(
+      @NotNull final Context context, @NotNull final ReadableMap rnOptions) {
+    startSdk.startWithOptions(context, rnOptions, null, logger);
+  }
+
+  /**
+   * Start the Native Android SDK with the provided options
+   *
+   * @param context Android Context
+   * @param options Map with options
+   */
+  public static void startWithOptions(
+      @NotNull final Context context, @NotNull final Map<String, Object> options) {
+    ReadableMap rnOptions = RNSentryMapConverter.mapToReadableMap(options);
+    startWithOptions(context, rnOptions);
+  }
+
+  /**
+   * Start the Native Android SDK with options from `sentry.options.json` configuration file
+   *
+   * @param context Android Context
+   */
+  public static void start(@NotNull final Context context) {
+    try {
+      JSONObject jsonObject = getOptionsFromConfigurationFile(context);
+      ReadableMap rnOptions = RNSentryMapConverter.jsonObjectToReadableMap(jsonObject);
```

Comment:
Our current lint doesn't enforce it, but let's use the `@NotNull/@Nullable` annotations here.

Let's add tests to ensure we gracefully continue if the file is not present or contains broken JSON and the init with configure callback is used.

If the `start()` is used I agree it make sense to fail as there is no other way to add the options.

Reply:
Good point 👍 
I've handled this in https://github.com/getsentry/sentry-react-native/pull/4474